src/coreclr/jit/llvm.cpp

```diff
@@ -721,11 +728,18 @@ void Llvm::emitDoNothingCall()
 
 void Llvm::buildAdd(GenTree* node, Value* op1, Value* op2)
 {
-    if (op1->getType()->isPointerTy() && op2->getType()->isIntegerTy())
+    Type* op1Type = op1->getType();
+    if (op1Type->isPointerTy() && op2->getType()->isIntegerTy())
     {
-        mapGenTreeToValue(node, _builder.CreateGEP(op1, op2));
+        // the second operator to gep is effectively the number of pointer contained types to add.
+        // For i8*, this is the same as the llvm add operator, but for i32* it would be 4 * add
+        // E.g.
+        // %27 = getelementptr %"[S.P.CoreLib]Internal.Runtime.MethodTable", %"[S.P.CoreLib]Internal.Runtime.MethodTable"* %1, i32 %3
+        // Does not add %3 to the *MethodTable, but adds %3 * sizeof(MethodTable)
+        // Force all pointers to i8* to get the IR GT_ADD semantics
+        mapGenTreeToValue(node, _builder.CreateGEP(castIfNecessary(op1, Type::getInt8PtrTy(_llvmContext)), op2));
```

Comment:
Feels like a losing battle, reconciling the fact pointers in IR are just integers with the fact they are not in LLVM...

Let's call this good enough for now (obviously, we could have `integer + pointer` here, but this is a bit of a best effort code anyway).

Nit: I would shorten the comment to something like `GEPs scale indices, bitcasting to i8* makes them equivalent to the raw offsets we have in IR`.